### DISCUSSION TITLE
how to NOT clear the screen on exit?

### DISCUSSION THREAD

**Discussion by kasperk81** [microsoft/terminal#14681] [Q&A] [answered]
exiting from a manually launched `wsl` shell instance from command prompt or powershell clears the screen.

can i stop it from happening? i have lost valuable outputs from last few commands ran in wsl and regretting typing `e x i t`.

also this is inconsistent behavior. in cmd this does not clear the screen
✅
```cmd
: start cmd sub-shell
> cmd
> dir
> exit
```
neither does:
✅
```cmd
: start powershell sub-shell
> powershell
> dir
> exit
```
but this does:
❌
```cmd
: start wsl sub-shell
> wsl
> dir
> exit
```
sub-sub-shell in wsl does not either:
✅
```cmd
: start wsl sub-shell
> wsl
> dir
: nested shell in wsl
> bash
> dir
: exit from sub-sub-shell
> exit
```
if user really wanted to clear the screen, they can with `cls` or `clear`, the 'force clear' behavior is bizarre.

**Comment by DHowett** [comment, marked as answer]
The default "bash logout" script that comes with Ubuntu (and a number of other distributions) clears the screen when the last shell exits to prevent the leakage of sensitive information.

You can change this behavior by removing (or editing) `~/.bash_logout`

Its default contents are as follows (on Ubuntu):

```bash
# ~/.bash_logout: executed by bash(1) when login shell exits.

# when leaving the console clear the screen to increase privacy

if [ "$SHLVL" = 1 ]; then
    [ -x /usr/bin/clear_console ] && /usr/bin/clear_console -q
fi
```
